dynamic_programming/minimum_edit_distance.cpp

```diff
@@ -0,0 +1,158 @@
+/**
+ * @file
+ * @brief Implementation of [Minimum Edit Distance](https://en.wikipedia.org/wiki/Edit_distance) using Dynamic Programing
+ *
+ * @details
+ *
+ * Given two strings str1 & str2 and we have to calculate the minimum
+ * number of operations (Insert, Remove, Replace) required to convert
+ * str1 to str2.
+ *
+ * ### Algorithm
+ *
+ * We will solve this problem using Naive recursion. But as we are
+ * approaching with a DP solution. So, we will take a DP array to
+ * store the solution of all sub-problems so that we don't have to
+ * perform recursion again and again. Now to solve the problem, We
+ * can traverse all characters from either right side of the strings
+ * or left side. Suppose we will do it from the right side. So, there
+ * are two possibilities for every pair of characters being traversed.
+ * 1. If the last characters of two strings are the same, Ignore
+ * the characters and get the count for the remaining string.
+ * So, we get the solution for lengths m-1 and n-1 in a DP array.
+ *
+ * 2. Else, (If last characters are not the same), we will consider all
+ * three operations (Insert, Remove, Replace) on the last character of
+ * the first string and compute the minimum cost for all three operations
+ * and take the minimum of three values in the DP array.
+ * For Insert: Recur for m and n-1
+ * For Remove: Recur for for m-1 and n
+ * For Replace: Recur for for m-1 and n-1
+ *
+ * @author [Nirjas Jakilim](github.com/nirzak)
+ */
+
+#include <cassert>     /// for assert
+#include <iostream>   /// for IO operations
+#include <vector>    /// for std::vector
+/**
+ * @namespace dynamic_programming
+ * @brief Dynamic Programming algorithms
+ */
+
+namespace dynamic_programming {
+
+/**
+ * @namespace Minimum Edit Distance
+ * @brief Implementation of [Minimum Edit Distance](https://en.wikipedia.org/wiki/Edit_distance) algorithm
+ */
+
+namespace minimum_edit_distance {
+
+/**
+ * @brief Takes input of the cost of
+ * three operations: Insert, Replace and Delete
+ * and return the minimum cost among them.
+ * @param x used to pass minimum cost of Insert operations
+ * @param y used to pass minimum cost of Replace operations
+ * @param z used to pass minimum cost of Delete operations
+ */
+
+int min(int x, int y, int z) {

```

Comment:
Consider using `uint64_t`, `uint32_t`, `uint16_t`, or even `uint8_t` for non-negative values. For negative values, use `int64_t` or their appropriate size ([reference](https://en.cppreference.com/w/cpp/types/integer)). Check out other parts of the code, too. 🙂

Reply:
> Consider using `uint64_t`, `uint32_t`, `uint16_t`, or even `uint8_t` for non-negative values. For negative values, use `int64_t` or their appropriate size ([reference](https://en.cppreference.com/w/cpp/types/integer)). Check out other parts of the code, too. 🙂

Doesn't it make the code more complex for beginners?

Reply:
`uint64_t` or `int64_t` (or their appropiate size) are better than `int`.